src/Rings/mpoly.jl

```diff
@@ -745,7 +749,7 @@ function singular_assure(I::IdealGens)
     I.gens.Sx = singular_poly_ring(I.Ox; keep_ordering = I.keep_ordering)
     I.gens.S = Singular.Ideal(I.Sx, elem_type(I.Sx)[I.Sx(x) for x = I.O])
   end
-  if I.isGB
+  if I.isGB && I.ord == monomial_ordering(I.gens.Ox, internal_ordering(I.gens.Sx))
```

Comment:
This was wrong: the polynomial ring on the Singular side might have a monomial ordering different from `I.ord`.